specs/experimental/fault-proof/stage-one/honest-challenger-fdg.md

```diff
@@ -132,17 +134,32 @@ def respond_claim(claim: Claim, correct_trace: List[ClaimHash, MAX_TRACE]) -> Re
     else:
         return Response.ATTACK
 
+def claim_response(claim: Claim, response: Response, trace: List[ClaimHash, MAX_TRACE]) -> Claim:
+    new_position = claim.position << 1 if Response.ATTACK else (claim.position+1) << 1
+    claim_hash = chal_trace{trace_index(new_position)]
+    return Claim(parent=claim, position=new_position, claim_hash=claim_hash)
+
+def is_freeloader(claim: Claim, team: Team, trace: List[ClaimHash, MAX_TRACE]) -> Claim:
+    if claim.parent is None:
+        return None
+    correct_claims = {claim_response(c, respond(c, team, trace), trace) for c in ALL_CLAIMS}
```

Comment:
this doesn't work.
There are two scenarios;
- regular dishonest claims is where we consider the path validity of dishonest claims
- freeloader mode where we ignore path validity.

When responding to a claim, we need to figure out which scenario we're dealing with and adapt the response logic appropriately. But first we need a more rigorous definition of a freeloader claim. One definition I'm working with is the following:

"Freeloaders are simply dishonest claims that are left-most positioned". This definition encompasses freeloaders (using the original definition) that switch teams as they counter other freeloaders and so on.
I'm only 60% confident that this is the correct definition. But we cannot proceed until this is proven.